src/main/java/com/foundation/search/controller/Controller.java

```diff
@@ -12,25 +12,136 @@
 
 package com.foundation.search.controller;
 
-import com.foundation.search.view.View;
+import com.foundation.search.model.Criteria;
 import com.foundation.search.model.Search;
+import com.foundation.search.model.SearchResult;
+import com.foundation.search.view.SearchView;
+import java.util.List;
 
-import java.awt.event.ActionEvent;
-
+/**
+ * This class integrates the view and model layers
+ *
+ * @author Mary Ricalde
+ * @version 1.0.
+ */
 public class Controller {
-    private View view;
+    private SearchView view;
     private Search search;
 
+    /**
+     * The constructor of Controller class where the view and model classes are
+     * instantiated and the action listener for the search button is called
+     */
     public Controller() {
-        this.view = new View();
+        this.view = new SearchView();
         this.search = new Search();
+        view.getMainPanel().getParametersPanel().getSearchButton().addActionListener(e -> getSearchCriteria());
+        view.getMainPanel().getParametersPanel().getCleanButton().addActionListener(c -> cleanResultTable());
     }
 
-    public void getBasicFields() {
+    /**
+     * This method reads the fields for the searching criteria and send these values
+     * to the search method.
+     * This set the list of files results to the table result
+     */
+    public void getSearchCriteria() {
+        Criteria searchCriteria = new Criteria("C:/");
```

Comment:
Please to replace the hard-coded path by some variable

Reply:
I removed the hard-coded path, the Criteria constructor doesn't require a variable to instantiate